src/api/app/models/bs_request.rb

```diff
@@ -927,26 +927,8 @@ def auto_accept
   end
 
   # Check if 'user' is maintainer in _all_ request targets:
-  def is_target_maintainer?(user = User.current)
-    has_target = false
-    is_target_maintainer = true
-    bs_request_actions.each do |a|
-      next unless a.target_project
-      if a.target_package
-        tpkg = Package.find_by_project_and_name(a.target_project, a.target_package)
-        if tpkg
-          has_target = true
-          is_target_maintainer &= user.can_modify?(tpkg)
-          next
-        end
-      end
-      tprj = Project.find_by_name(a.target_project)
-      if tprj
-        has_target = true
-        is_target_maintainer &= user.can_modify?(tprj)
-      end
-    end
-    has_target && is_target_maintainer
+  def is_target_maintainer?(user)
+    bs_request_actions.map { |action| action.is_target_maintainer?(user) }.all?
```

Comment:
wennschon dennschon:smiley:
```ruby
 bs_request_actions.all? { |action| action.is_target_maintainer?(user) }
```

Reply:
Hmm, yesterday I tried and was somehow led to believe that .all? would return false on empty array. Today it does not - which is a problem

Reply:
well, it's not - as we have no requests without actions. just the old `has_target` logic looked for it. But an action would target will return false on is_target_action? (I guess I could write specs for that)

Reply:
done that now - sorry for the early morning brabling :)